app/services/moves/import_people.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+module Moves
+  class ImportPeople
+    attr_accessor :items
+
+    def initialize(items)
+      self.items = items
+    end
+
+    def call
+      import_people
+    end
+
+  private
+
+    def import_people
+      people_nomis_prison_numbers = items.map { |item| item.fetch(:person_nomis_prison_number) }
+      people = NomisClient::People.get(people_nomis_prison_numbers).map { |p| [p.fetch(:prison_number), p] }.to_h
+      alerts = NomisClient::Alerts.get(people_nomis_prison_numbers).group_by { |p| p.fetch(:offender_no) }
+      personal_care_needs = NomisClient::PersonalCareNeeds
+                            .get(people_nomis_prison_numbers)
+                            .group_by { |p| p.fetch(:offender_no) }
+
+      new_person_count = 0
+      changed_profile_count = 0
+      people.each do |offender_no, person_data|
```

Comment:
About this rather convoluted and complex loop, my *******PERSONAL TASTE******** (:D) is to extract in external methods everything I can, using the writes to the db as some kind of guidance to have each write in a separate method. That normally also guide me through naming the method properly by greatly limiting its scope.

Reply:
yes, this smells a lot, we need to refactor it (not at the moment! 😄 )